src/app.js

```diff
@@ -45,115 +50,86 @@ export function app(props) {
         map.call(element.childNodes, function(element) {
           return element.nodeType === 3
             ? element.nodeValue
-            : hydrate(element, map)
+            : elementToNode(element, map)
         })
       )
     )
   }
 
-  function createActions(actions, withActions, lastPath) {
-    Object.keys(withActions || {}).map(function(name) {
-      return typeof withActions[name] === "function"
-        ? (actions[name] = function(data) {
-            return typeof (data = withActions[name](
-              getPath(lastPath, appState),
-              getPath(lastPath, appActions),
-              data
-            )) === "function"
+  function init(module, state, actions) {
+    if (module.init) {
+      callbacks.push(function() {
+        module.init(state, actions)
+      })
+    }
+
+    assign(state, module.state)
+
+    initActions(state, actions, module.actions)
+
+    Object.keys(module.modules || {}).map(function(i) {
```

Comment:
```js
for (i in module.modules || {}) {
```

Using it here will save some bytes and give us better perf, I guess.

Reply:
We will need a function to wrap the expression anyway because we would be accessing a shared variable `i` inside the new action definition `source[i]`. If we could use `let` instead of `var` it would work though.